endpoint.bal

```diff
@@ -0,0 +1,510 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/url;
+
+# Twitter Client object.
+#
+# + clientId - The consumer key of the Twitter account
+# + clientSecret - The consumer secret of the Twitter account
+# + accessToken - The access token of the Twitter account
+# + accessTokenSecret - The access token secret of the Twitter account
+# + twitterClient - HTTP Client endpoint
+@display {label: "Twitter API Client"}
+public client class  Client {
+    
+    string clientId;
+    string clientSecret;
+    string accessToken;
+    string accessTokenSecret;
+
+    http:Client twitterClient;
+
+    public isolated function init(@display {label: "Connection Configuration"} TwitterConfiguration twitterConfig) returns error? {
+        self.twitterClient = check new(TWITTER_API_URL, twitterConfig.clientConfig);
+        self.clientId = twitterConfig.clientId;
+        self.clientSecret = twitterConfig.clientSecret;
+        self.accessToken = twitterConfig.accessToken;
+        self.accessTokenSecret = twitterConfig.accessTokenSecret;
+    }
+
+    # Update the user's Tweet.
+    #
+    # + tweetText - The text of tweet to update
+    # + updateStatusOptions - Options for tweet update
+    # + return - If success, returns Status object, else returns error.
+    @display {label: "Post a Tweet"}
+    remote function tweet(@display {label: "Tweet text"} string tweetText, @display {label: "Optional Update Options"} UpdateStatusOptions? updateStatusOptions = ()) 
+                          returns @tainted @display {label: "Status"} Status|error {
+        http:Request request = new;
+
+        string mediaIds = "";
+        string attachmentUrl = "";
+
+        if (updateStatusOptions?.media_ids is string) {
+            mediaIds = <string>updateStatusOptions?.media_ids;
+        }
+        if (updateStatusOptions?.attachment_url is string) {
+            attachmentUrl = <string>updateStatusOptions?.attachment_url;
+        }
+
+        string resourcePath = TWEET_ENDPOINT;
+        string encodedStatusValue = check url:encode(tweetText, UTF_8);
+        string urlParams = STATUS + encodedStatusValue + "&";
```

Comment:
'&' is used in many places.
Shall we add it as a constant?

Reply:
Added constant